src/lib/reasoners/satml.ml

```diff
@@ -579,26 +579,6 @@ module Make (Th : Theory.S) : SAT_ML with type th = Th.t = struct
     a.var.reason <- None;
     a.var.vpremise <- []
 
-  let enqueue_assigned env (a : Atom.atom) =
-    if Options.get_debug_sat () then
-      Printer.print_dbg "[satml] enqueue_assigned: %a@." Atom.pr_atom a;
-    if a.neg.is_guard then begin
-      (* if the negation of a is (still) a guard, it should be forced to true
-         during the first decisions.
-         If the SAT tries to deduce that a.neg is true (ie. a is false),
-         then we have detected an inconsistency. *)
-      assert (a.var.level <= env.next_dec_guard);
-      (* guards are necessarily decided/propagated before all other atoms *)
-      raise (Unsat None);
```

Comment:
I'm writing a test for it now ;)

Reply:
This code is here to ensure we never propagate the negation of a guard (you give scenarios where this can happen), but in `enqueue_assigned` it was useless because `enqueue_assigned` is only called on literals that were *already* propagated, and thus would have hit the corresponding path in `enqueue` at the moment they were initially propagated.